### PR TITLE
close all servers of agent properly

### PR DESCRIPTION
**Pull request by ramaraochavali**
- [ ] Ambient
- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [X] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

**Review by lei-tang — APPROVED** [review]

File: pilot/cmd/pilot-agent/app/cmd.go [hunk 1]
```diff
@@ -173,6 +173,7 @@ func newProxyCommand() *cobra.Command {
 				return err
 			}
 			wait()
+			agent.Close()
```
Comment (howardjohn) [inline pilot/cmd/pilot-agent/app/cmd.go:176]: do we want defer so it closes on errors as well?

**Review by howardjohn — COMMENTED** [review]
would be useful to have more context on the user facing impacts?

**Comment by ramaraochavali** [comment]
> would be useful to have more context on the user facing impacts?

There is no user facing impact. I was debugging draining behaviour and stumbled on this.

File: pilot/cmd/pilot-agent/app/cmd.go (hunk 1 above)
Comment (ramaraochavali) [reply pilot/cmd/pilot-agent/app/cmd.go:176]: Makes sense. Let me change

**Review by hzxuzhonghu — APPROVED** [review]

**Pull request merged**